src/core/mac/mac_frame.hpp

```diff
@@ -757,16 +788,25 @@ class Frame: public otRadioFrame
      * @returns A pointer to the char string buffer.
      *
      */
-    const char *ToInfoString(char *aBuf, uint16_t aSize);
+    const char *ToInfoString(char *aBuf, uint16_t aSize) const;
 
 private:
-    uint8_t *FindSequence(void);
-    uint8_t *FindDstPanId(void);
-    uint8_t *FindDstAddr(void);
-    uint8_t *FindSrcPanId(void);
-    uint8_t *FindSrcAddr(void);
-    uint8_t *FindSecurityHeader(void);
+    enum
+    {
+        kInvalidIndex = 0xff,
+        kSequenceIndex = kFcfSize,
+    };
+
+    uint16_t GetFrameControlField(void) const;
+    uint8_t FindDstPanIdIndex(void) const;
```

Comment:
Yes, I came up this because I think it could be a common topic in C++ and OpenThread, i.e. dealing with const and non-const member functions. We might encounter it here and there. In section `Item 3: Use const whenever possible` of *Effective C++*, it recommends using `const_cast` to avoid duplication in const and non-const member functions. However, I'm good with the current implementation. Thanks for your explanation.